ckanext/spectrum/actions.py

```diff
@@ -29,6 +30,22 @@ def dataset_duplicate(context, data_dict):
     return toolkit.get_action('package_show')(context, {'id': duplicate_dataset['id']})
 
 
+@toolkit.chained_action
+def package_create(next_action, context, data_dict):
+    dataset_type = data_dict.get('type', '')
+
+    valid_types = toolkit.get_action("scheming_dataset_schema_list")(context, {})
+    valid_types_list = ", ".join(valid_types)
+
+    if 'dataset' not in valid_types_list:
+        valid_types_list += ", dataset"
```

Comment:
I'm not sure if this is right, but I wanted to have something so in case of no schemas at least 'dataset' would be here. Originally in ckan, when someone creates a dataset with empty/nonexistent type a default one is assigned and saved in db - namely 'dataset'.